naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeResourceField.scala

```diff
@@ -25,72 +41,84 @@ object NaptimeResourceField extends StrictLogging {
-      resourceName: String): OptionType[DataMap] = {
+      resourceName: ResourceName): Either[SchemaError, OptionType[DataMap]] = {
     val resource = schemaMetadata.getResource(resourceName)
-    val schema = schemaMetadata.getSchema(resource).getOrElse {
-      throw SchemaGenerationException(s"Cannot find schema for $resourceName")
-    }
+    schemaMetadata.getSchema(resource).map { schema =>
 
-    val resourceObjectType = OptionType(ObjectType[SangriaGraphQlContext, DataMap](
-      name = formatResourceName(resource),
-      description = schema.getDoc,
-      fieldsFn = () => {
-        Option(schema.getFields).map(_.asScala.map { field =>
-          generateField(field, schemaMetadata, resource, schema)
-        }.toList).getOrElse(List.empty).flatten
-      }))
-    resourceObjectType
+      val resourceObjectType = OptionType(
+        ObjectType[SangriaGraphQlContext, DataMap](
+          name = formatResourceName(resource),
+          description = schema.getDoc,
+          fieldsFn = () => {
+            Option(schema.getFields).map(
+              _.asScala.map { field =>
+                generateField(field, schemaMetadata, resource, schema)
+              }.toList).getOrElse(List.empty).flatten
+          }))
+      resourceObjectType
+    }.toRight(SchemaNotFound(resourceName))
   }
 
   private[schema] def generateField(
       field: RecordDataSchema.Field,
       schemaMetadata: SchemaMetadata,
       resource: Resource,
       schema: RecordDataSchema): Option[Field[SangriaGraphQlContext, DataMap]] = {
-    val forwardRelationOption = field.getProperties.asScala.get(Relations.PROPERTY_NAME).map(_.toString)
+
+    val forwardRelationOption = field.getProperties.asScala
+      .get(Relations.PROPERTY_NAME)
+      .map(_.toString)
+
     if (forwardRelationOption.isDefined) {
-      val relatedResource = schemaMetadata.getResource(forwardRelationOption.get)
-      if (relatedResource.handlers.exists(_.kind == HandlerKind.MULTI_GET)) {
-        Some(FieldBuilder.buildField(schemaMetadata, field, Option(schema.getNamespace),
-          resourceName = formatResourceName(resource)))
-      } else {
-        logger.warn(s"Unable to build field for forward relation from " +
-          s"${resource.name} -> ${forwardRelationOption.get} due to the lack of a MULTI_GET handler")
-        None
-      }
+      (for {
+        resourceName <- ResourceName.parse(forwardRelationOption.get)
+        relatedResource <- schemaMetadata.getResourceOpt(resourceName)
+      } yield {
+        if (relatedResource.handlers.exists(_.kind == HandlerKind.MULTI_GET)) {
+          Some(FieldBuilder.buildField(schemaMetadata, field, Option(schema.getNamespace),
+            resourceName = ResourceName.fromResource(resource)))
+        } else {
+          logger.warn(s"Unable to build field for forward relation from " +
+            s"${resource.name} -> ${forwardRelationOption.get} due to the lack of a MULTI_GET handler")
+          None
```

Comment:
i'm actually going to be deprecating this entire section (forward relations aren't used anymore), so instead of refactoring things to get this to work, i'm just going to add a todo to delete the whole section :)